modules/os2forms_fasit/src/Form/SettingsForm.php

```diff
@@ -0,0 +1,233 @@
+<?php
+
+namespace Drupal\os2forms_fasit\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\os2forms_fasit\Helper\CertificateLocatorHelper;
+use Drupal\os2forms_fasit\Helper\Settings;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\OptionsResolver\Exception\ExceptionInterface as OptionsResolverException;
+
+/**
+ * Organisation settings form.
+ */
+final class SettingsForm extends FormBase {
+  use StringTranslationTrait;
+
+  public const FASIT_API_BASE_URL = 'fasit_api_base_url';
+  public const FASIT_API_TENANT = 'fasit_api_tenant';
+  public const FASIT_API_VERSION = 'fasit_api_version';
+  public const CERTIFICATE = 'certificate';
+
+  /**
+   * Constructor.
+   */
+  public function __construct(private readonly Settings $settings, private readonly CertificateLocatorHelper $certificateLocatorHelper) {
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): SettingsForm {
+    return new static(
+      $container->get(Settings::class),
+      $container->get(CertificateLocatorHelper::class)
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'os2forms_fasit_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @phpstan-param array<string, mixed> $form
+   * @phpstan-return array<string, mixed>
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+
+    $fasitApiBaseUrl = $this->settings->getFasitApiBaseUrl();
+    $form[self::FASIT_API_BASE_URL] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Fasit API base url'),
+      '#required' => TRUE,
+      '#default_value' => !empty($fasitApiBaseUrl) ? $fasitApiBaseUrl : NULL,
+      '#description' => $this->t('Specifies which base url to use. This is disclosed by Schultz'),
+    ];
+
+    $fasitApiTenant = $this->settings->getFasitApiTenant();
+    $form[self::FASIT_API_TENANT] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Fasit API tenant'),
+      '#required' => TRUE,
+      '#default_value' => !empty($fasitApiTenant) ? $fasitApiTenant : NULL,
+      '#description' => $this->t('Specifies which tenant to use. This is disclosed by Schultz'),
+    ];
+
+    $fasitApiVersion = $this->settings->getFasitApiVersion();
+    $form[self::FASIT_API_VERSION] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Fasit API version'),
+      '#required' => TRUE,
+      '#default_value' => !empty($fasitApiVersion) ? $fasitApiVersion : NULL,
+      '#description' => $this->t('Specifies which api version to use. Should probably be v2'),
+    ];
+
+    $certificate = $this->settings->getCertificate();
```

Comment:
We should real use the key module for this type of data.

Reply:
Which have support for different vault backends.

Reply:
Support for keys will be added when it had been finished! 

Currently, it is till work in progress.